std/range/package.d

```diff
@@ -6629,10 +6630,10 @@ if (isForwardRange!RangeOfRanges &&
         }
     }
 
-    // ElementType opIndex(size_t n)
-    // {
-    //     return _input[n].front;
-    // }
+    ElementType opIndex(size_t n)
```

Comment:
Cool! Let's do this simpler implementation:

```d
auto ref opIndex(size_t n) { return _input[n]; }
```

Then you can remove the import and the alias.

Reply:
I would throw const into the mix:
```
auto ref opIndex(size_t n) const { return _input[n]; }

```

Reply:
const would unnecessarily limit what can be done and the ranges this could work with - it's more restrictive than C++'s.

Reply:
`inout` then?

Reply:
`inout` has a problem here, because the `_input` type's `opIndex` must support `inout` in order for it to work properly. Otherwise, it would not work with a range that supplies only mutable `opIndex`.

Regardless, you need a static check around `opIndex` to make sure the underlying range supports it.